index.js

```diff
@@ -284,3 +284,27 @@ export function escapePath(path) {
 
 	return path.replace(/[\\.[]/g, '\\$&');
 }
+
+function * deepKeysIterator(object, currentPath) {
+	for (const [key, value] of Object.entries(object)) {
+		const property = typeof currentPath === 'undefined' ? escapePath(key) : `${currentPath}.${escapePath(key)}`;
```

Comment:
The way I usually implement this kind of thing is making `currentPath` a non-optional array, which handles the ‘empty’ case slightly more nicely. This line would become `const nextPath = [...currentPath, escapePath(key)]`. and the yield statements `yield * deepKeysIterator(value, newPath)` and `yield newPath.join('.')`